File: 7 - 10001st prime/primes.jl

function isPrime(x)
    if x == 1
        return false
    elseif x < 4
        return true
    elseif x%2 == 0
        return false
    elseif x < 9
        return true
    elseif x%3 == 0
        return false
    else
        r = floor(sqrt(x))
        f = 5
        while f <= r
            if x%f == 0 || x%(f+2) == 0
                return false
            end
            f = f + 6
        end
        return true
    end
end

function index(x)
    index = 1
    i = 1
    while index != x
        if isPrime(i)
            index += 1
            #println(index, ", ", i)
        end
        i += 2
    end
    return i - 2
end

println(index(10001))

File: 7 - 10001st prime/primes.lua
require("math")
local prime
prime = function(x)
  if x == 1 then
    return false
  end
  for i = 2, math.sqrt(x) do
    if x % i == 0 then
      return false
    end
  end
  return true
end
local index
index = function(x)
  local j, i = 0, 1
  while j < x do
    i = i + 1
    if prime(i) == true then
      j = j + 1
    end
  end
  return i
end
return print(index(10001))
